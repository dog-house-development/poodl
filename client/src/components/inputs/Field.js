import React, { Component } from "react";
import PropTypes from "prop-types";

const propTypes = {
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string,
  id: PropTypes.string.isRequired,
  type: PropTypes.string,
  size: PropTypes.oneOf(["normal", "large"]),
  content: PropTypes.string,
  placeholder: PropTypes.string,
  label: PropTypes.string,
  error: PropTypes.object
};

const defaultProps = {
  size: "normal",
  type: "text"
};

class Field extends Component {
  constructor(props) {
    super(props);
    this.state = { value: "" };
    this.handleChange = this.handleChange.bind(this);
  }

  onComponentDidMount() {
    this.setState({ value: this.props.content });
  }

  handleChange(e) {
    e.preventDefault();
    this.setState({ value: e.target.value });
    this.props.onChange(e);
  }

  render() {
    return (
      <div>
        {this.props.label}
        <input
          id={this.props.id}
          type={this.props.type}
          name={this.props.name}
          value={this.state.value}
          placeholder={this.props.placeholder}
          onChange={this.handleChange}
          className={`field ${this.props.size}`}
          error={this.props.error}
        />
      </div>
    );
  }
}

Field.propTypes = propTypes;
Field.defaultProps = defaultProps;
export default Field;
