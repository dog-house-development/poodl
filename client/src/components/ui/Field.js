import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import _ from 'lodash';

const propTypes = {
    onChange: PropTypes.func.isRequired,
    name: PropTypes.string,
    id: PropTypes.string.isRequired,
    type: PropTypes.string,
    min: PropTypes.string,
    max: PropTypes.string,
    size: PropTypes.oneOf(['normal', 'large']),
    value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
    kind: PropTypes.oneOf(['primary', 'secondary']),
    placeholder: PropTypes.string,
    label: PropTypes.string,
    autoComplete: PropTypes.oneOf(['on', 'off']),
    autoFocus: PropTypes.oneOf(['on', 'off']),
    spellCheck: PropTypes.oneOf(['true', 'false']),
    sidebyside: PropTypes.oneOf([1, 2]),
    error: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
    onClick: PropTypes.func,
    present: PropTypes.bool,
    clearable: PropTypes.bool,
    leftIcon: PropTypes.string
};

const defaultProps = {
    size: 'normal',
    type: 'text',
    autoComplete: 'on',
    value: '',
    kind: 'primary'
};

class Field extends Component {
    constructor(props) {
        super(props);
        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(e) {
        e.preventDefault();
        this.props.onChange(e);
    }

    getClearMarkup() {
        if (this.props.clearable && !_.isEmpty(this.props.value)) {
            return (
                <i
                    className={classnames('material-icons', 'clear-icon', this.props.size, this.props.kind)}
                    onClick={this.onClearClick}>
                    clear
                </i>
            );
        }
    }

    onClearClick = () => {
        const e = { target: { value: '' } };
        this.props.onChange(e);
    };

    getLeftIconMarkup() {
        if (this.props.leftIcon) {
            return <i className={classnames('material-icons', 'left-icon', this.props.size)}>{this.props.leftIcon}</i>;
        }
    }

    render() {
        const { style, present, editable, clearable, leftIcon, ...inputProps } = this.props;
        if (present) {
            return (
                <div style={style} className="field-wrapper editable-field-wrapper">
                    <p className="field-label">{this.props.label}</p>
                    <p>{this.props.value}</p>
                </div>
            );
        }

        return (
            <div
                style={style}
                className={classnames('field-wrapper', this.props.className, {
                    'inline-field': this.props.sidebyside
                })}>
                <p className="field-label">{this.props.label}</p>
                <div className="field-outer">
                    {this.getLeftIconMarkup()}
                    <input
                        {...inputProps}
                        className={classnames(
                            'field',
                            this.props.size,
                            this.props.kind,
                            { 'first-side-by-side-input': this.props.sidebyside === 1 },
                            { 'second-side-by-side-input': this.props.sidebyside === 2 },
                            { 'field-error-border': this.props.error },
                            { 'clearable-field': this.props.clearable },
                            { 'include-left-icon': this.props.leftIcon }
                        )}
                    />
                    {this.getClearMarkup()}
                    <p className="field-error-label">{this.props.error}</p>
                </div>
            </div>
        );
    }
}

Field.propTypes = propTypes;
Field.defaultProps = defaultProps;
export default Field;
