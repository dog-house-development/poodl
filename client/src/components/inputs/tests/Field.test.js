import React from "react";
import ReactDOM from "react-dom";
import Field from "../Field";

it("renders without crashing", () => {
  const props = {
    onChange: () => {},
    id: "name"
  };
  const div = document.createElement("div");
  ReactDOM.render(<Field {...props} />, div);
  ReactDOM.unmountComponentAtNode(div);
});
