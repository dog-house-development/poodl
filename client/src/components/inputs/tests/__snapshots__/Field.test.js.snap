// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field tests render should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Field
    content=""
    error={Object {}}
    id="email"
    label="Email:"
    name="email"
    onChange={[Function]}
    placeholder="Enter email..."
    size="normal"
    type="email"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Email:",
        <input
          className="field normal"
          error={Object {}}
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Enter email..."
          type="email"
          value=""
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      "Email:",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "field normal",
          "error": Object {},
          "id": "email",
          "name": "email",
          "onChange": [Function],
          "placeholder": "Enter email...",
          "type": "email",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Email:",
          <input
            className="field normal"
            error={Object {}}
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Enter email..."
            type="email"
            value=""
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Email:",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "field normal",
            "error": Object {},
            "id": "email",
            "name": "email",
            "onChange": [Function],
            "placeholder": "Enter email...",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
